apps/desktop/src/lib/backend/updater.ts

```diff
@@ -38,37 +39,51 @@ export class UpdaterService {
 	readonly version = derived(this.update, (update) => update?.version);
 
 	prev: Update | undefined;
-	unlistenFn: any;
+	unlistenStatusFn: any;
+	unlistenManualCheckFn: any;
 	intervalId: any;
 
 	constructor() {}
 
 	private async start() {
 		const currentVersion = await getVersion();
 		this.currentVersion.set(currentVersion);
-		await this.checkForUpdate();
-		setInterval(async () => await this.checkForUpdate(), 3600000); // hourly
-		this.unlistenFn = await onUpdaterEvent((status) => {
+		this.unlistenManualCheckFn = listen<string>('menu://global/update/clicked', () => {
+			this.checkForUpdate(true);
+		});
+
+		this.unlistenStatusFn = await onUpdaterEvent((status) => {
 			const err = status.error;
 			if (err) {
 				showErrorToast(err);
 				posthog.capture('App Update Status Error', { error: err });
 			}
 			this.status.set(status.status);
 		});
+
+		await this.checkForUpdate();
+		setInterval(async () => await this.checkForUpdate(), 3600000); // hourly
 	}
 
 	private async stop() {
-		this.unlistenFn?.();
+		this.unlistenStatusFn?.();
+		this.unlistenManualCheckFn?.();
 		if (this.intervalId) {
 			clearInterval(this.intervalId);
 			this.intervalId = undefined;
 		}
 	}
 
-	async checkForUpdate() {
-		const update = await checkUpdate();
-		if (update.manifest) {
+	async checkForUpdate(isManual = false) {
+		const update = await Promise.race([
+			checkUpdate(), // In DEV mode this never returns.
+			new Promise<UpdateResult>((resolve) =>
+				setTimeout(() => resolve({ shouldUpdate: false }), 30000)
```

Comment:
@PavelLaptev in dev mode you need to wait for this timeout until "up to date" should be displayed. In prod, the checkUpdate function above will also return `shouldUpdate: false` if no new binary is available.